TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Resources/InverseKinematics.java

```diff
@@ -50,112 +50,129 @@ public class InverseKinematics {
-    private final double armExtensionMax  = 19.5, armExtensionMin  = 0.0;
+    private final double ARM_EXTENSION_MIN = 0.0, ARM_EXTENSION_MAX = 19.5;
 
-    private final double armRotationMin   = 8.0, armRotationMax   = 180.0;
+    private final double ARM_ROTATION_MIN = 8.0, ARM_ROTATION_MAX = 180.0;
 
-    private final double elbowRotationMin = -180.0, elbowRotationMax = 180.0;
+    private final double ELBOW_ROTATION_MIN = -180.0, ELBOW_ROTATION_MAX = 180.0;
 
-    private final double twistMin         = 0, twistMax         = 180.0;
+    private final double TWIST_MIN = 0, TWIST_MAX = 180.0;
     
     //the distance between the center of the robot and the center of the claw on the x axis
-    private final double elbowOffset = 4.0;
+    private final double ELBOW_OFFSET = 4.0;
     
     // the length in inches of the non-extending shaft.
-    private final double elbowLength = 11.18;
+    private final double ELBOW_LENGTH = 11.18;
 
     // how far the pivot point of the arm is away from the center of the robot.
-    private final double pivotOffset = 2.55;
+    private final double PIVOT_OFFSET = 2.55;
 
     // the distance between the bottom of the wheels to the center of the arm pivot point
-    private final double chassisHeight = 5.73;
-    
-    
+    private final double CHASSIS_HEIGHT = 5.73;
 
-    private final double tubeLength = 16.75;
+    private final double TUBE_LENGTH = 16.75;
 
     // how close we can be to a sample and still pick it up without moving back = ~21.5
-    private final double minL = pivotOffset + Math.sqrt(tubeLength*tubeLength- Math.pow(elbowLength + 1.5 - chassisHeight,2));
+    private final double MIN_H = PIVOT_OFFSET + Math.hypot(TUBE_LENGTH, ELBOW_LENGTH + 1.5 - CHASSIS_HEIGHT);
     
     // how far away a sample can be without us breaking the expansion limit
-    private final double maxL = 26; // inches from center the of the robot
+    private final double MAX_H = 26; // inches from center the of the robot
     
     
     // the constructor for this class.... that needs to do nothing.. yup
     public InverseKinematics() {
-
+        /* Suggestion: Have constructor take robot position and sample position as arguments
+        move the calculate kinematics functionality to the constructor.
+        (But not the twist)
+         */
     }
 
-    //TODO: re-comment main code as I did it wrong the first time - Tobin
-
-    
-    
     public boolean verifyLength(double Rx,double Ry,double Tx,double Ty){
-        double l = calculateDistance(Tx,Ty,Rx,Ry)-pivotOffset;
-      
-      return !(l + pivotOffset > maxL);
-//        !(l < minL) &&
+        //l is equal to h-pivot offset
+        double h = Math.hypot(Tx-Rx,Ty-Ry);
+
+      return h <= MAX_H;
+
     }
     /**
      * The function you call from the main code to calculate the position of the robot.
      * This function decides which side you are trying to grab from then calls the corresponding method
```

Comment:
This comment is no longer true as we have made a universal algorithm instead of a very strict one.

Reply:
Good point. We will update comments throughout.